pkg/model/iam.go

```diff
@@ -112,35 +112,42 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 
 		// Generate additional policies if needed, and attach to existing role
-		if b.Cluster.Spec.AdditionalPolicies != nil {
-			roleAsString := reflect.ValueOf(role).String()
-			additionalPolicies := *(b.Cluster.Spec.AdditionalPolicies)
-
-			if additionalPolicy, ok := additionalPolicies[strings.ToLower(roleAsString)]; ok {
-				additionalPolicyName := "additional." + name
-
-				{
-					p := &iam.IAMPolicy{
-						Version: iam.IAMPolicyDefaultVersion,
-					}
-
-					statements := make([]*iam.IAMStatement, 0)
-					json.Unmarshal([]byte(additionalPolicy), &statements)
-					p.Statement = append(p.Statement, statements...)
-
-					policy, err := p.AsJSON()
-					if err != nil {
-						return fmt.Errorf("error building IAM policy: %v", err)
-					}
-
-					t := &awstasks.IAMRolePolicy{
-						Name:           s(additionalPolicyName),
-						Role:           iamRole,
-						PolicyDocument: fi.WrapResource(fi.NewStringResource(policy)),
-					}
-					c.AddTask(t)
+		{
+			additionalPolicy := ""
+			if b.Cluster.Spec.AdditionalPolicies != nil {
+				roleAsString := reflect.ValueOf(role).String()
+				additionalPolicies := *(b.Cluster.Spec.AdditionalPolicies)
+
+				additionalPolicy = additionalPolicies[strings.ToLower(roleAsString)]
+			}
+
+			additionalPolicyName := "additional." + name
+
+			t := &awstasks.IAMRolePolicy{
+				Name: s(additionalPolicyName),
+				Role: iamRole,
+			}
+
+			if additionalPolicy != "" {
+				p := &iam.IAMPolicy{
+					Version: iam.IAMPolicyDefaultVersion,
+				}
+
+				statements := make([]*iam.IAMStatement, 0)
+				json.Unmarshal([]byte(additionalPolicy), &statements)
+				p.Statement = append(p.Statement, statements...)
+
+				policy, err := p.AsJSON()
+				if err != nil {
+					return fmt.Errorf("error building IAM policy: %v", err)
 				}
+
+				t.PolicyDocument = fi.WrapResource(fi.NewStringResource(policy))
+			} else {
+				t.PolicyDocument = fi.WrapResource(fi.NewStringResource(""))
```

Comment:
Nice spot - that was the problem that was causing the e2e to fail!  I pushed the fix (ignore the "notFound" error on delete) as an additional commit...